src/game/Protocol/WorldSocket.cpp

```diff
@@ -170,54 +261,68 @@ int WorldSocket::HandleAuthSession(WorldPacket& recvPacket)
-    std::unique_ptr<QueryResult> result(LoginDatabase.PQuery("SELECT a.`id`, aa.`gmLevel`, a.`sessionkey`, a.`last_ip`, a.`v`, a.`s`, a.`mutetime`, a.`locale`, a.`os`, a.`platform`, a.`flags`, "
-    //      11         12                13
-        "a.`email`, a.`email_verif`, ab.`unbandate` > UNIX_TIMESTAMP() OR ab.`unbandate` = ab.`bandate` FROM `account` a LEFT JOIN `account_access` aa ON a.`id` = aa.`id` AND aa.`RealmID` IN (-1, %u) "
-        "LEFT JOIN `account_banned` ab ON a.`id` = ab.`id` AND ab.`active` = 1 WHERE a.`username` = '%s' && DATEDIFF(NOW(), a.`last_login`) < 1 ORDER BY aa.`RealmID` DESC LIMIT 1", realmID, safe_account.c_str()));
+    auto accountQueryResult =
+            LoginDatabase.PQuery("SELECT "
+                                 "a.`id`, "             // 0
+                                 "aa.`gmLevel`, "       // 1
+                                 "a.`sessionkey`, "     // 2
+                                 "a.`last_ip`, "        // 3
+                                 "a.`v`, "              // 4
+                                 "a.`s`, "              // 5
+                                 "a.`mutetime`, "       // 6
+                                 "a.`locale`, "         // 7
+                                 "a.`os`, "             // 8
+                                 "a.`platform`, "       // 9
+                                 "a.`flags`, "          // 10
+                                 "a.`email`, "          // 11
+                                 "a.`email_verif`, "    // 12
+                                 "ab.`unbandate` > UNIX_TIMESTAMP() OR ab.`unbandate` = ab.`bandate` " // 13
+                                 "FROM `account` a "
+                                 "LEFT JOIN `account_access` aa ON a.`id` = aa.`id` AND aa.`RealmID` IN (-1, %u) "
+                                 "LEFT JOIN `account_banned` ab ON a.`id` = ab.`id` AND ab.`active` = 1 WHERE a.`username` = '%s' && DATEDIFF(NOW(), a.`last_login`) < 1 "
+                                 "ORDER BY aa.`RealmID` DESC LIMIT 1", realmID, safe_account.c_str());
 
     // Stop if the account is not found
-    if (!result)
+    if (!accountQueryResult)
     {
         packet.Initialize(SMSG_AUTH_RESPONSE, 1);
         packet << uint8(AUTH_UNKNOWN_ACCOUNT);
 
         SendPacket(packet);
 
         sLog.Out(LOG_BASIC, LOG_LVL_ERROR, "WorldSocket::HandleAuthSession: Sent Auth Response (unknown account).");
-        return -1;
+        return HandlerResult::Fail;
     }
 
-    Field* fields = result->Fetch();
+    Field* fields = accountQueryResult->Fetch();
 
     // Prevent connecting directly to mangosd by checking
     // that same ip connected to realmd previously.
-    if (strcmp(fields[3].GetString(), GetRemoteAddress().c_str()) &&
-        serverAddressList.find(GetRemoteAddress()) == serverAddressList.end())
+    if (fields[3].GetCppString() == GetRemoteIpString() && serverAddressList.find(GetRemoteIpString()) == serverAddressList.end())
```

Comment:
wrong condition, should be only if the ips differ, not if they are the same

if (fields[3].GetCppString() != GetRemoteIpString() && serverAddressList.find(GetRemoteIpString()) == serverAddress